src/Microsoft.OData.Client/BulkUpdateSaveResult.cs

```diff
@@ -53,15 +84,31 @@ internal BulkUpdateGraph BulkUpdateGraph
             get { return this.bulkUpdateGraph; }
         }
 
-        protected override Stream ResponseStream => throw new NotImplementedException();
+        /// <summary>
+        /// Returns an instance of the link descriptors.
+        /// </summary>
+        internal Dictionary<Descriptor, LinkDescriptor> LinkDescriptors
+        {
+            get { return this.linkDescriptors; }
+        }
+
+        protected override Stream ResponseStream
+        {
+            get { return this.responseStream; }
+        }
 
         /// <summary>
-        /// Syncronous bulk update request.
+        /// Synchronous bulk update request.
         /// </summary>
         /// <typeparam name="T"> The type of the top-level objects to be deep-updated.</typeparam>
         /// <param name="objects"> The top-level objects of the type to be deep updated.</param>
         internal void BulkUpdateRequest<T>(params T[] objects)
         {
+            if (objects == null || objects.Length == 0)
+            {
+                throw Error.Argument(Strings.Util_EmptyArray, "top-level objects");

```

Comment:
Should we be using resources for "top-level objects"?

Reply:
@ElizabethOkerio You misunderstand my comment here. I meant the literal "top-level objects" should be moved to a resource files here such that it can be accessed using `Strings.<RESOURCE_ID>`